pages/map/index.tsx

```diff
@@ -1,6 +1,148 @@
+}   
+
+interface MapPageContent{
+    children: React.ReactNode
+}
+
+const GMap = ({center, children}: GMap) => {
+    //Creates Google Map
+    const defaultCoords: Coords = {lat: 40.7128, lng: -74.0060}; //Defaults to NYC coords
+    const mapOptions = {
+        mapTypeControl: true,
+        streetViewControl: true
+    }
+
+    return(
+        <div className='shadow' style={{width: "100%", height: "80vh"}}>
+            {center.lat && center.lng ? 
+                <GoogleMapReact
+                    yesIWantToUseGoogleMapApiInternals
+                    bootstrapURLKeys={{key: mapsKey, libraries:[]}}
+                    defaultCenter={defaultCoords}
+                    center={center}
+                    defaultZoom={11}
+                    options={mapOptions}
+                >
+                    <BiStreetView
+                        //@ts-ignore
+                        lat={center.lat}
+                        lng={center.lng}
+                        text="Marker"
+                        size={26}
+                        style={{position:'absolute', transform: 'translate(-50%, -50%)'}}
+                    />
+
+                    {children}
+                </GoogleMapReact>
+                : 
+                <GoogleMapReact
+                    bootstrapURLKeys={{key: mapsKey, libraries:[]}}
+                    defaultCenter={defaultCoords}
+                    defaultZoom={11}
+                    options={mapOptions}
+                >
+                </GoogleMapReact>
+            }
+        </div>
+    )
+}
+
+const MapPageContent = ({children}: MapPageContent) => {
+    const [zip, setZip] = useState('');
+    const [userCoords, setUserCoords] = useState({lat: null, lng: null});
+
+    const changeHandler = (e: any): void => {
+        //Sets zip code state from form value
+        e.preventDefault();
+
+        //@ts-ignore
+        setZip(e.target.value);
+    }
+
+    const getUserCoords = (e: any): void => {
+        //Sets user coordinates as state
+        e.preventDefault();
+
+        if(navigator.geolocation){
+            navigator.geolocation.getCurrentPosition((position) => {
+                let userCoords = {
+                    lat: position.coords.latitude,
+                    lng: position.coords.longitude
+                }
+
+                setUserCoords((prev: any) => {
+                    return {
+                        ...prev,
+                        lat: userCoords.lat,
+                        lng: userCoords.lng
+                    }
+                });
+            });
+        }
+    }
+
+    return(
+        <Container>
+            {children}
+
+            <Row className='mb-2'>
+                <Col>
+                    <Form className='d-flex flex-row justify-content-between' onSubmit={e => e.preventDefault()}>
+                        <Form.Control required type='zip' placeholder='Enter Zip Code...' value={zip} onChange={(e: any) => changeHandler(e)}></Form.Control>
```

Comment:
this line seems kind of long. generally, it's best practice to limit to 80 characters. btw, you'll find other sources of text like news articles are often formatted with narrow paragraphs for the same reason. it's harder to read wide blocks of text.

https://www.codereadability.com/maximum-line-length/#:~:text=Other%20code%20standards%20in%20the,have%20to%20follow%20it%20religiously.

a random article I just found that discusses this: https://link.springer.com/chapter/10.1007/11555261_59